.github/workflows/blocks-playwright.yml

```diff
@@ -0,0 +1,66 @@
+name: Run Blocks Playwright Tests
+on:
+    pull_request:
+        paths:
+            - 'plugins/woocommerce-blocks/**'
+            - 'plugins/woocommerce/src/Blocks/**'
+            - 'plugins/woocommerce/templates/**'
+            - 'plugins/woocommerce/patterns/**'
+
+jobs:
+    PlaywrightE2ETests:
+        name: Playwright E2E tests - ${{ matrix.config.name }}
+        timeout-minutes: 60
+        runs-on: ubuntu-latest
+        defaults:
+            run:
+                working-directory: plugins/woocommerce-blocks
+        strategy:
+            matrix:
+                config:
+                    [
+                        {
+                            name: Normal,
+                            file: playwright.config.ts,
+                            resultPath: test-results,
+                        },
+                        {
+                            name: Classic,
+                            file: playwright.classic-theme.config.ts,
+                            resultPath: test-results-classic-theme,
+                        },
+                        {
+                            name: SideEffects,
+                            file: playwright.side-effects.config.ts,
+                            resultPath: test-results-side-effects,
+                        },
+                    ]
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Setup WooCommerce Monorepo
+              uses: ./.github/actions/setup-woocommerce-monorepo
+              with:
+                  install: '@woocommerce/plugin-woocommerce...'
+                  build: '@woocommerce/plugin-woocommerce'
+
+            - name: Install Playwright
+              run: pnpm --filter='@woocommerce/block-library' exec playwright install --with-deps
+
+            - name: Start wp-env
+              run: pnpm --filter='@woocommerce/block-library' wp-env start
+
+            - name: Configure the environment
+              run: pnpm wp-env:config
```

Comment:
@lanej0, we use different configs for Playwright. Playwright configs live in `./tests/e2e/bin` (in Woo Blocks). First we should update the mapping in `plugins/woocommerce-blocks/.wp-env.json`:

```diff
"env": {
	"tests": {
		"mappings": {
			"wp-content/mu-plugins": "./node_modules/@wordpress/e2e-tests/mu-plugins",
			"wp-content/plugins/gutenberg-test-plugins": "./node_modules/@wordpress/e2e-tests/plugins",
			"wp-cli.yml": "./wp-cli.yml",
			"custom-plugins": "./tests/e2e/mocks/custom-plugins",
-			"wp-content/plugins/woocommerce/blocks-bin": "./bin"
+			"wp-content/plugins/woocommerce/blocks-bin": "./bin",
+			"wp-content/plugins/woocommerce/blocks-bin/playwright": "./tests/e2e/bin"
		}
	}
},
```

Then we should fix the path in `plugins/woocommerce-blocks/tests/e2e/bin/test-env-setup.sh`

```diff
#!/usr/bin/env bash

# Extract the relative path from the plugin root to this script directory.
# By doing so, we can run this script from anywhere.
script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
head_dir=$(cd "$(dirname "$script_dir")" && cd ../../.. && pwd)
relative_path=${script_dir#$head_dir/}

# Run the main script in the container for better performance.
-wp-env run tests-cli -- bash wp-content/plugins/$relative_path/scripts/index.sh
+wp-env run tests-cli -- bash wp-content/plugins/woocommerce/blocks-bin/playwright/scripts/index.sh
```

Finally, let's update the `env:start` script in `plugins/woocommerce-blocks/package.json`

```diff
-"env:start": "pnpm run wp-env start && ./tests/e2e/bin/test-env-setup.sh",
+"env:start": "pnpm --filter='@woocommerce/block-library' wp-env start && ./tests/e2e/bin/test-env-setup.sh",
```
`.github/workflows/blocks-playwright.yml` can be updated as such:

```diff
steps:
            - uses: actions/checkout@v3

            - name: Setup WooCommerce Monorepo
              uses: ./.github/actions/setup-woocommerce-monorepo
              with:
                  install: '@woocommerce/plugin-woocommerce...'
                  build: '@woocommerce/plugin-woocommerce'

            - name: Install Playwright
              run: pnpm --filter='@woocommerce/block-library' exec playwright install --with-deps

-          - name: Start wp-env
+          - name: Start wp-env and configure the environment
+            run: pnpm --filter='@woocommerce/block-library' env:start
-            run: pnpm --filter='@woocommerce/block-library' wp-env start

-            - name: Configure the environment
-              run: pnpm wp-env:config
```